dev/docgen/mapDir.ts

```diff
@@ -35,13 +41,24 @@ export const mapDir = (
                         transformedOutputPath
                     )
                 }
-                return [transformedOutputPath, transformedContents]
+                return [
+                    transformedOutputPath,
+                    transformedContents,
+                    repoRelativePath
+                ]
             })
     )
+    const mappedContents = []
     for (const target of options.targets) {
         rmSync(target, { recursive: true, force: true })
-        for (const [path, contents] of fileContentsByRelativeDestination) {
+        for (const [
+            path,
+            contents,
+            source
+        ] of fileContentsByRelativeDestination) {
             const resolvedPath = join(target, path)
+            mappedContents.push({ source, generated: resolvedPath })
+            writeJson(fromHere(".docgenSourceMap.json"), mappedContents)
```

Comment:
This file should be written to each mapped directory, and should look like this:

```json
{
    "type.ts": "src/__snippets__/type.ts",
    "space.ts": "src/__snippets__/space.ts"
}
```